django/urls/resolvers.py

```diff
@@ -243,48 +243,49 @@ def __str__(self):
     r"<(?:(?P<converter>[^>:]+):)?(?P<parameter>[^>]+)>"
 )
 
+whitespace_set = frozenset(string.whitespace)
 
+
+@functools.lru_cache
 def _route_to_regex(route, is_endpoint):
     """
     Convert a path pattern into a regular expression. Return the regular
     expression and a dictionary mapping the capture names to the converters.
     For example, 'foo/<int:pk>' returns '^foo\\/(?P<pk>[0-9]+)'
     and {'pk': <django.urls.converters.IntConverter>}.
     """
-    original_route = route
     parts = ["^"]
+    all_converters = get_converters()
     converters = {}
-    while True:
-        match = _PATH_PARAMETER_COMPONENT_RE.search(route)
-        if not match:
-            parts.append(re.escape(route))
-            break
-        elif not set(match.group()).isdisjoint(string.whitespace):
+    previous_end = 0
+    for match in _PATH_PARAMETER_COMPONENT_RE.finditer(route):
```

Comment:
Changed `match` to `match_` to avoid using a [keyword](https://docs.python.org/3.12/reference/lexical_analysis.html#soft-keywords).

Reply:
Fair. It's a soft keyword deliberately to not break use of regexes like this, as I heard recently on https://podcasts.apple.com/gb/podcast/core-py/id1712665877?i=1000647675451 😉